### PR TITLE
Default Storage Key Creator Tests.

### PR DESCRIPTION
**Pull request by SHeimlich**
Add tests for the DefaultStorageKeyCreator class as part of the unit tests for Allocator.

### CODE REVIEW

**Comment by SHeimlich** [comment]
\gcbrun

File: javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  @Before
+  open fun setUp() = runBlocking {
+    RamDisk.clear()
+    DriverAndKeyConfigurator.configure(null)
+  }
+
+  // Simple Case.
+  @Test
+  fun keyCreator_createStorageKeysIfNecessary() = creatorKeyTest(PersonPlan)
```
Comment (shans) [inline javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt:30]: maybe add the plans as comments to the top of each test, so its easy to see what's going on?

File: javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  // Complex case.
+  @Test
+  fun keyCreator_createStorageKeysIfNecessary_manyParticles() = creatorKeyTest(MultiplePersonPlan)
+
+  /**
+   * Given a [Plan], test that it's keys start as [CreatableStorageKey]s and are transformed
```
Comment (shans) [inline javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt:37]: it's -> its

File: javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+   * using the [DefaultStorageKeyCreator.createStorageKeysIfNecessary] method. Verify that the plan
+   * returned
```
Comment (shans) [inline javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt:39]: Verify that the plan returned what?

File: javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+   */
+  private fun creatorKeyTest(plan: Plan) {
+    val idGenerator = Id.Generator.newSession()
+    val testArcId = idGenerator.newArcId("Test")
+    val keyCreator = DefaultStorageKeyCreator()
+
+    // Verify the handle storage keys start as creatable.
```
Comment (shans) [inline javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt:46]: The tests above shouldn't be called "IfNecessary" if we're enforcing that all storage keys start as creatable; I'm guessing the name comes from the method name under test? In which case, we should be testing examples where not every key needs to be created too (I acknowledge we're sort of doing this here by calling createStorageKeysIfNecessary twice, but a plan with a mixture of creatable and non-creatable keys isn't currently tested).

File: javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt (hunk 1 above)
Comment (SHeimlich) [reply javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt:39]: The recipes are rather long (especially for MultiplePersonPlan) so I noted where to find the Arcs Manifest File.

File: javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt (hunk 3 above)
Comment (SHeimlich) [reply javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt]: Oops, apparently my train of thought just trailed off. Updated the comment.

File: javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt (hunk 4 above)
Comment (SHeimlich) [reply javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt:110]: Added a mixed test as well.

File: javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt (hunk 1 above)
Comment (shans) [reply javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt:39]: I don't think 24 lines (including empty lines) is too long, particularly if it helps with understanding of the test. Then again maybe that's not the case here.

File: javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt (hunk 4 above)
Comment (shans) [reply javatests/arcs/core/allocator/DefaultStorageKeyCreatorTest.kt:110]: you should actually be able to create plans that have pre-created storage keys (e.g. if they reference storage from other plans).

**Review by shans — APPROVED** [review]

**Review by shans — APPROVED** [review]

**Pull request closed without merging**